### PR TITLE
E2E tests: restore plugin update test

### PR DESCRIPTION
**Pull request by adimoldovan**
#### Changes proposed in this Pull Request:

Restore a refactored Jetpack plugin update test.

Also:
- included in CI in a separate job, to reduce the time required with setup/teardown.
- also created the `update-flow.sh` script, which can be used to test the update via CLI.
- added `clean` and `new` commands in env management script.
- other minor fixes and changes

#### Jetpack product discussion
p9dueE-4c4-p2#comment-7071

#### Does this pull request change what data or activity we track or use?
no

#### Testing instructions:
* Plugin update test is green.

### CODE REVIEW

**Review by anomiex — COMMENTED** [review]
Seems ok at a read through, and the test runs to completion.

A few suggestions inline. The one in prepare-update.sh I'd really like to see addressed. The other one I could easily be convinced to leave for a follow-up PR.

File: projects/plugins/jetpack/tests/e2e/bin/update/prepare-update.sh [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+VERSION="99.9-alpha"
+
+printf "\nDeactivating Jetpack and removing any related plugins\n"
+wp plugin --allow-root deactivate jetpack || true
+rm -rf /var/www/html/wp-content/plugins/jetpack || true
+rm -rf /var/www/html/wp-content/plugins/boost || true
+rm -rf /var/www/html/wp-content/plugins/beta || true
+rm -rf /var/www/html/wp-content/plugins/debug-helper || true
+rm -rf /var/www/html/wp-content/plugins/backup || true
+rm -rf /var/www/html/wp-content/plugins/vaultpress || true
```
[the remark this answers is not in the document]
Comment (adimoldovan) [reply projects/plugins/jetpack/tests/e2e/bin/update/prepare-update.sh]: That's a good idea! Addressed by [25a283c](https://github.com/Automattic/jetpack/pull/22738/commits/25a283cc659b2023d280e9acae8288ef5b6c2724)

File: projects/plugins/jetpack/tests/e2e/bin/update/prepare-zip.sh [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+if [[ "$OSTYPE" == "linux-gnu"* ]]; then
+	# assuming Debian
+	if [[ -z $(which zip) ]]; then sudo apt-get update && sudo apt-get install -qy zip; fi
+fi
+
+printf "\nPreparing zip file\n"
+mkdir -p "$TMP_DIR"
+cd "$BASE_PATH/../../../../.."
+find -L jetpack ! -path '**/node_modules/*' ! -path '**/\.cache/*' ! -path '**/tests/*' ! -path '**/changelog/*' ! -path '**/wordpress/*' ! -path '**/\.idea/*' -print | zip -q "$ZIP_FILE" -@
```
[the remark this answers is not in the document]
Comment (adimoldovan) [reply projects/plugins/jetpack/tests/e2e/bin/update/prepare-zip.sh:24]: Extracting that into `tools/includes` would make sense. I will address this in a follow up though.

**Review by anomiex — APPROVED** [review]
Let's do it.

**Pull request merged**